octave: abort when printing figure to pdf

## Issue description
When saving a plot figure to pdf in Octave, the program panics and aborts

### Steps to reproduce
```
$ octave --no-gui
octave:1> t = linspace(0,1);
octave:2> x = power(abs(t), 1/3);
octave:3> plot(t,x);
octave:4> print -dpdf output.pdf
warning: print.m: epstool binary is not available.
Some output formats are not available.
warning: called from
    __print_parse_opts__ at line 385 column 9
    print at line 315 column 8
error: gl2ps_print: support for gl2ps was unavailable or disabled when Octave was built
terminate called after throwing an instance of 'octave::execution_exception'
panic: Aborted -- stopping myself...
attempting to save variables to 'octave-workspace'...
save to 'octave-workspace' complete
[1]    14326 abort      octave --no-gui
```
Reproducible with both `octave` and `octaveFull` from `nixos-unstable`. If the GUI is used, the program exits with signal 6.

## Technical details
```
 - system: `"x86_64-linux"`
 - host os: `Linux 4.9.78, NixOS, 17.09.2873.4c3c0e824aa (Hummingbird)`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 1.11.16`
 - channels(root): `"nixos-17.09.2873.4c3c0e824aa"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs`
```

--- a/maintainers/maintainer-list.nix
+++ b/maintainers/maintainer-list.nix
@@ -5362,6 +5362,15 @@
     github = "twey";
     name = "James ‘Twey’ Kay";
   };
+  twhitehead = {
+    name = "Tyson Whitehead";
+    email = "twhitehead@gmail.com";
+    github = "twhitehead";
+    keys = [{
+      longkeyid = "rsa2048/0x594258F0389D2802";
+      fingerprint = "E631 8869 586F 99B4 F6E6  D785 5942 58F0 389D 2802";
+    }];
+  };
   typetetris = {
     email = "ericwolf42@mail.com";
     github = "typetetris";

--- a/pkgs/development/libraries/gl2ps/default.nix
+++ b/pkgs/development/libraries/gl2ps/default.nix
@@ -1,0 +1,29 @@
+{ stdenv, fetchurl, cmake
+, zlib, libGL, libGLU, libpng, freeglut }:
+
+stdenv.mkDerivation rec {
+  version = "1.4.0";
+  name = "gl2ps-${version}";
+
+  src = fetchurl {
+    url = "http://geuz.org/gl2ps/src/${name}.tgz";
+    sha256 = "1qpidkz8x3bxqf69hlhyz1m0jmfi9kq24fxsp7rq6wfqzinmxjq3";
+  };
+
+  buildInputs = [
+    cmake
+    zlib
+    libGL
+    libGLU
+    libpng
+    freeglut
+  ];
+
+  meta = with stdenv.lib; {
+    homepage = http://geuz.org/gl2ps;
+    description = "An OpenGL to PostScript printing library";
+    platforms = platforms.all;
+    license = licenses.lgpl2;
+    maintainers = with maintainers; [raskin twhitehead];
+  };
+}

--- a/pkgs/top-level/all-packages.nix
+++ b/pkgs/top-level/all-packages.nix
@@ -3280,6 +3280,8 @@ in
 
   gitea = callPackage ../applications/version-management/gitea { };
 
+  gl2ps = callPackage ../development/libraries/gl2ps { };
+
   glusterfs = callPackage ../tools/filesystems/glusterfs { };
 
   glmark2 = callPackage ../tools/graphics/glmark2 { };

